src/Core/src/MauiApp.cs

```diff
@@ -1,15 +1,19 @@
 using System;
-using System.Collections.Generic;
 
 namespace Microsoft.Maui
 {
 	public abstract class MauiApp : App
 	{
-		public abstract IWindow CreateWindow(IActivationState state);
-
-		public MauiApp()
+		protected MauiApp()
 		{
+			if (Current != null)
+				throw new InvalidOperationException($"Only one {nameof(App)} instance is allowed.");
 
+			Current = this;
 		}
+
+		public static MauiApp? Current { get; internal set; }
```

Comment:
I moved `Current` into `MauiApp` instead of the base `App` as this is more for users, not the internal framework. I feel that users expect a way to get a "current" app, but if this is not the case, then we go full DI and if the user wants the app, then the user can fetch it from the services. But then this sort of requires an opt-in to ctor injection as there is no static services.